dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp

```diff
@@ -77,6 +78,22 @@ CommandBarFlyoutCommandBar::CommandBarFlyoutCommandBar()
             {
                 if (owningFlyout.AlwaysExpanded())
                 {
+
+                    // The below IsOpen(true) call ends up setting the visual state to one of the open ones, making
+                    // the shadow stay visible as it's closing. Remove the shadow manually in this specific scenario.
+                    if (SharedHelpers::Is21H1OrHigher())
+                    {
+                        winrt::IControlProtected thisAsControlProtected = *this;
+                        auto grid = GetTemplateChildT<winrt::Grid>(L"OuterOverflowContentRoot", thisAsControlProtected);
+                        if (winrt::IUIElement10 grid_uiElement10 = grid)
+                        {
+                            if (grid_uiElement10.Shadow())
+                            {
+                                grid_uiElement10.Shadow(nullptr);
+                            }
+                        }
+                    }
```

Comment:
Did it not work to just prevent us from going to the visual state in question?  You generally don't want to mix setting properties programmatically and setting properties in visual states if it can be avoided, as one can sometimes stomp on another to create unexpected results.

Reply:
Yeah, if there is a way to avoid this, we should. can we set a flag that is used to block the IsOpen handler from going to the offending state?

---
In reply to: [612708086](https://github.com/microsoft/microsoft-ui-xaml/pull/4743#discussion_r612708086) [](ancestors = 612708086)

Reply:
I originally tried the flag, which didn't seem to fix the issue so I went with this.

After playing around with the flag a bit more, I'm not so sure if the problem is that it's setting the wrong visual state, because the flag successfully blocks it from going to any other state except "Default"/"Collapsed". However, it does still seem like the `IsOpen(true)` call causes the issue as it works fine if I remove it. I'm just not sure anymore what side effect it has to cause the issue.

Reply:
I talked to Leon offline - it looks like the issue is that we IsOpen is causing us not to revert back to the default state, which is where we remove the shadow.  I suggested to him that he could instead remove the shadow in the Closing storyboard.  This does technically have us with two visual states writing to the same property, but the Closing storyboard will always result in the flyout no longer being drawn, so it will always "win" as it'll be the last storyboard to be played before the flyout disappears.  After opening and closing the flyout multiple times with that change, things look good - the shadow now disappears when it should on closing and appears when it should on opening.